.gitignore

```diff
@@ -1,2 +1,5 @@
 cpp/build
 qa/logs
+.cache
```

Comment:
Let's keep the project .gitignore to the files which are actually generated by the project build. Stuff like `.vscode` should be handled in a global .gitignore.